### PR TITLE
Support explain for OV models

### PR DESCRIPTION
**Pull request by GalyaZalesskaya**
### Summary

Change `engine/explain` to process OV model
Update `tests/integration/api/test_engine_api.py` to cover explain for Torch and IR models

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [x] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by negvet — DISMISSED** [review]

File: tests/integration/api/test_engine_api.py [hunk 1]
```diff
@@ -71,3 +71,31 @@ def test_engine_from_config(
...
+    # Predict Torch model with explain
+    predictions = engine.predict(explain=True)
+    assert len(predictions[0].saliency_maps) > 0
+
+    # Export IR model with explain
+    exported_model_with_explain = engine.export(explain=True)
+    assert exported_model_with_explain.exists()
+
+    # Infer IR Model with explain
+    if task in OVMODEL_PER_TASK:
```
Comment (sovrasov) [inline tests/integration/api/test_engine_api.py]: How the `task` variable is used here?

File: tests/integration/api/test_engine_api.py [hunk 2]
```diff
@@ -71,3 +71,31 @@ def test_engine_from_config(
...
+        # Predict
+        predictions = engine.predict(checkpoint=exported_model_with_explain, accelerator="cpu")
+        assert len(predictions) > 0
+        # Explain
+        explain_results = engine.explain(checkpoint=exported_model_with_explain)
+        assert len(explain_results[0].saliency_maps) > 0
```
Comment (sovrasov) [inline tests/integration/api/test_engine_api.py]: It'd be perfect, if predict and explain give the same saliency maps

File: tests/integration/api/test_engine_api.py (hunk 1 above)
Comment (GalyaZalesskaya) [reply tests/integration/api/test_engine_api.py]: Thank you for noticing, fixed

File: tests/integration/api/test_engine_api.py (hunk 2 above)
Comment (GalyaZalesskaya) [reply tests/integration/api/test_engine_api.py]: Done

**Review by sovrasov — DISMISSED** [review]

**Review by negvet — DISMISSED** [review]

File: src/otx/engine/engine.py [hunk 3]
```diff
@@ -572,29 +572,47 @@ def explain(
...
+            model=model,
             optimizer=self.optimizer,
             scheduler=self.scheduler,
         )
-        if datamodule is None:
-            datamodule = self.datamodule
         lit_module.label_info = datamodule.label_info
 
+        # NOTE, trainer.test takes only lightning based checkpoint.
+        # So, it can't take the OTX1.x checkpoint.
```
Comment (harimkang) [inline src/otx/engine/engine.py]: As i know, we can load 1.X checkpoint with this code.
@sungmanc Is it right?

File: src/otx/engine/engine.py [hunk 4]
```diff
@@ -572,29 +572,47 @@ def explain(
...
-        if explain_config is None:
-            explain_config = ExplainConfig()
+        model = self.model
+
+        if checkpoint is not None:
+            checkpoint = str(checkpoint)
+        elif self.checkpoint is not None:
+            checkpoint = str(self.checkpoint)
+        else:
+            checkpoint = None
```
Comment (harimkang) [inline src/otx/engine/engine.py]:
```suggestion
        checkpoint = checkpoint if checkpoint is not None else self.checkpoint
```

File: src/otx/engine/engine.py (hunk 3 above)
Comment (sungmanc) [reply src/otx/engine/engine.py]: Yes, we could load the OTX1.X checkpoint.

Comment (GalyaZalesskaya) [reply src/otx/engine/engine.py]: Thank you, removed this comment

File: src/otx/engine/engine.py (hunk 4 above)
Comment (GalyaZalesskaya) [reply src/otx/engine/engine.py]: Updated, thank you

**Review by harimkang — DISMISSED** [review]

**Review by sovrasov — DISMISSED** [review]

**Review by harimkang — DISMISSED** [review]

**Review by harimkang — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Review by negvet — APPROVED** [review]

**Pull request merged**